### ISSUE TITLE
Simultaneous context operations causes crash on AMD hardware

### ISSUE THREAD

**Issue by binary1248** [LaurentGomila/SFML#778]
If user code makes SFML access any pixel format or OpenGL context functions simultaneously from multiple threads, the application might crash on AMD hardware.

See: http://en.sfml-dev.org/forums/index.php?topic=12895.0

Test with:
```cpp
#include <SFML/Graphics.hpp>
#include <deque>
#include <iostream>

void threadFunc(int i)
{
    sf::Context context;

    std::cout << "Context " << i << " created.\n";

    for(;;)
    {
        sf::Texture texture;
        texture.create(100, 100);
    }
}

int main()
{
    sf::RenderWindow window(sf::VideoMode(200, 200), "Test");

    std::deque<sf::Thread*> threads;

    for (int i = 0; i < 40; i++)
    {
        threads.push_back(new sf::Thread(threadFunc, i));
        threads.back()->launch();
    }

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }

    while (!threads.empty())
    {
        threads.front()->terminate();
        delete threads.front();
        threads.pop_front();
    }
}
```
If it doesn't crash on the first attempt, run it multiple times.

**Comment by eXpl0it3r** [comment]
Can not confirm on my AMD Radeon HD 7500M/7600M card.

**Comment by MarioLiebisch** [comment]
Probably with an AMD CPU rather than a GPU? Or both?

**Comment by binary1248** [comment]
Maybe it wasn't obvious enough, but this is an OpenGL context management issue, so completely unrelated to CPU :grin:. Besides... I've never heard of a specific CPU model causing any specific problems... Those kinds of things died out in the last millennium :wink:.

**Comment by MarioLiebisch** [comment]
You can never be sure with AMD. :D

**Comment by eXpl0it3r** [comment]
Has been fixed through #779
